### PR TITLE
Only load post body additional content when needed

### PR DESCRIPTION
**Pull request by enahum**

### CODE REVIEW

File: app/components/post_body/post_body.js [hunk 1]
```diff
@@ -66,6 +67,18 @@ class PostBody extends PureComponent {
         toggleSelected: emptyFunction
     };
 
+    constructor(props) {
+        super(props);
+
+        this.link = extractFirstLink(props.message);
```
Comment (jarredwitt) [inline app/components/post_body/post_body.js]: What are the benefits of not setting this in the state?

Comment (enahum) [reply app/components/post_body/post_body.js]: Avoid unnecesary re-render

Comment (hmhealey) [reply app/components/post_body/post_body.js]: Won't it only change when the message has already changed though? It won't cause a re-render on its own. Plus we're using it in the render function, so it should be in the state to keep it pure

**Comment by enahum** [comment]
@hmhealey @jarredwitt changes made

**Pull request merged**